src/mui/field/ReferenceArrayField.js

```diff
@@ -0,0 +1,106 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import LinearProgress from 'material-ui/LinearProgress';
+import get from 'lodash.get';
+
+// `crudGetOneReference` uses `crudGetMany` and deboucing
+// its usually more performant on `<List>`
+import { crudGetOneReference as crudGetOneReferenceAction } from '../../actions/dataActions';
+
+/**
+ * A container component that fetches records from another resource specified
+ * by an array of *ids* in current record.
+ *
+ * You must define the fields to be passed to the iterator component as children.
+ *
+ * @example Display all the products of the current order as datagrid
+ * <ReferenceArrayField label="Products" reference="products" source="product_ids">
+ *     <Datagrid>
+ *         <TextField source="id" />
+ *         <TextField source="description" />
+ *         <NumberField source="price" options={{ style: 'currency', currency: 'USD' }} />
+ *         <EditButton />
+ *     </Datagrid>
+ * </ReferenceArrayField>
+ *
+ * @example Display all the categories of the current product as a list of chips
+ * <ReferenceArrayField label="Categories" reference="categories" source="category_ids">
+ *     <SingleFieldList>
+ *         <ChipField source="name" />
+ *     </SingleFieldList>
+ * </ReferenceArrayField>
+ *
+ */
+export class ReferenceArrayField extends Component {
+    componentDidMount() {
+        this.fetchReferences();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.reference !== nextProps.reference ||
+            this.props.record.id !== nextProps.record.id) {
+            this.fetchReferences(nextProps);
+        }
+    }
+
+    fetchReferences({ crudGetOneReference, reference, ids } = this.props) {
+        ids.map(id => crudGetOneReference(reference, id));
```

Comment:
Argh! Not good. this will issue n requests no matter what. You should dispatch a single action, probably `crudGetMany`, or create a new one. It's the job of the `restClient` to map that to one or several HTTP requests, according to the abilities of the REST API.